tests/farmer_harvester/test_third_party_harvesters.py

```diff
@@ -0,0 +1,441 @@
+
+    await wait_until_node_type_connected(farmer.server, NodeType.FULL_NODE)
+    await wait_until_node_type_connected(farmer.server, NodeType.HARVESTER)  # Should already be connected
+    await wait_until_node_type_connected(full_node_1.server, NodeType.FULL_NODE)
+
+    fee_threshold = 0.5
+    max_fee_proofs = 5
+    fee_count = 0
+    proof_count = 0
+
+    farmer_reward_puzzle_hash = decode_puzzle_hash("txch1psqeaw0h244v5sy2r4se8pheyl62n8778zl6t5e7dep0xch9xfkqhx2mej")
+
+    async def intercept_farmer_new_proof_of_space(*args: Any) -> None:
+        nonlocal farmer
+        nonlocal fee_threshold
+        nonlocal max_fee_proofs
+        nonlocal proof_count
+        nonlocal fee_count
+        nonlocal farmer_reward_puzzle_hash
+
+        request: harvester_protocol.NewProofOfSpace = harvester_protocol.NewProofOfSpace.from_bytes(args[0])
+
+        fee_threshold_int = uint32(int(0xFFFFFFFF * fee_threshold))
+
+        fee_quality = calculate_harvester_fee_quality(request.proof.proof, request.challenge_hash)
+        if fee_quality <= fee_threshold_int and fee_count < max_fee_proofs:
+            fee_count += 1
+            request = dataclasses.replace(
+                request,
+                farmer_reward_address_override=farmer_reward_puzzle_hash,
+                fee_info=ProofOfSpaceFeeInfo(applied_fee_threshold=fee_threshold_int),
+            )
+
+        if proof_count <= max_fee_proofs:
+            proof_count += 1
+
+        peer: WSChiaConnection = args[1]
+        await FarmerAPI.new_proof_of_space(farmer.server.api, request, peer)
+
+    mocker.patch.object(farmer.server.api, "new_proof_of_space", side_effect=intercept_farmer_new_proof_of_space)
+
+    log_text_len = 0
+
+    def log_has_new_text() -> bool:
+        nonlocal caplog
+        nonlocal log_text_len
+
+        text_len = len(caplog.text)
+        if text_len > log_text_len:
+            log_text_len = text_len
+            return True
+
+        return False
+
+    # Load test data
+    blocks: List[FullBlock]
+    signage_points: SPList
+
+    (blocks, signage_points) = load_test_data()
+    assert len(blocks) == 1
+    await add_test_blocks_into_full_node(blocks, full_node_2)
+
+    # Inject signage points
+    await inject_signage_points(signage_points, full_node_1, full_node_2)
+
+    # Wait until we've received all the proofs
+    def received_all_proofs() -> bool:
+        nonlocal max_fee_proofs
+        nonlocal fee_count
+
+        return fee_count >= max_fee_proofs
+
+    await time_out_assert(60 * 60, received_all_proofs, True)
+
+    # Wait for the farmer to pick up the last proofs
+    await asyncio.sleep(2)
+
+    assert fee_count > 0
+    await time_out_assert(60, log_has_new_text, True)
+
+    find_message = "Fee threshold passed for challenge"
+    find_index = 0
+    log_text = caplog.text
+    fail_count = 0
+
+    for _ in range(fee_count):
+        index = log_text.find(find_message, find_index) + len(find_message)
+        if index < 0:
```

Comment:
it seems `index` is never < 0 in the test. is that OK?

Reply:
Will fix